### PR TITLE
36 build assertion3

### LINKED ISSUE #36: Build Ruleset 3: User commented on id-hub Discord channel which will generate a Verifiable Credential (VC)
Build **Ruleset 3: User commented on id-hub Discord channel** which will generate a **Verifiable Credential (VC)** if the following validates:
- Check the IDHub controller (user) has commented using his “ID-Hubber” role Discord user account
- **Important:** The specific Discord account must be provided and passed to the Parachain from the F/E using the `params` extrinsic param to verify this:
    - The Discord account must have the role of “ID-Hubber”
    - The Discord account must have posted 1+ messages within **id-hub** Discord channel
    - If the ruleset validates then a VC should be generated and returned to the F/E.

### PR DESCRIPTION
**Pull request by BillyWooo**
resolves #36

### CODE REVIEW

File: litentry/core/assertion-build/src/a2.rs [hunk 1]
```diff
@@ -23,79 +23,89 @@ extern crate sgx_tstd as std;
...
 
 	// TODO:
 	// generate_vc(who, identity, ...)
 
+	// Assign ID-Hubber role:
+	let post_path = format!(
+		"/discord/assgin/idhubber?guildid={:?}&handler={:?}",
+		guild_id.into_inner(),
+		handler.into_inner()
+	);
```
Comment (Kailai-Wang) [inline litentry/core/assertion-build/src/a2.rs]: Should the assignment of `IDHubber` be triggered by our assertion-build code or by F/E after it verifies VC? I thought that was a subsequent action after VC is verified.

IMO our assertion-build should only verify or compute things, not trigger any actions -- this should be done by the VC consumer. From an architectural PoV these should be separated.

Let's clarify it with @jingleizhang and maybe IDHub colleagues. This is unrelated to this PR though.

I'm happy with the rest, PR approved, thanks.

**Review by Kailai-Wang — APPROVED** [review]

File: litentry/core/assertion-build/src/a2.rs (hunk 1 above)
Comment (BillyWooo) [reply litentry/core/assertion-build/src/a2.rs]: About the IDHubber role assignment, it's defined in #35 . 
There should two parts hold this role info: F/E and bot. It's only the bot here.
Let's have a short alignment tomorrow morning.

Comment (deprecated1234) [reply litentry/core/assertion-build/src/a2.rs]: Based on the definition of assertion2 (https://github.com/litentry/tee-worker/issues/35), there are two ways for us to let the others(e.g, Bot) know about the verification results:

1.  TEE triggers the action to others under/behind assertion-build flow
2. user presents a vc to others after the assertion-build flow

Let's have an alignment today morning about:
- whether these trigger mechanisms should be contained in TEE or not
- in assertion-build flow,  whether the codes here should trigger/notify external endpoints or not

Comment (BillyWooo) [reply litentry/core/assertion-build/src/a2.rs]: Let use solution 2:
 1. TEE only serves the computation; 
 2. VC will be sent to F/E; 
 3. F/E trigger IDHubber role assignment;
 4. F/E align with bot.

**Review by ziming-zung — APPROVED** [review]
Looks good

**Pull request merged**